includes/class-wc-gateway-ppec-cart-handler.php

```diff
@@ -415,28 +415,44 @@ public function maybe_enqueue_checkout_js( $widget_title, $widget_instance = arr
 	 */
 	protected function get_button_settings( $settings, $context = '' ) {
 		$prefix = $context ? $context . '_' : $context;
-		$data = array(
+		$data   = array(
 			'button_layout'        => $settings->{ $prefix . 'button_layout' },
 			'button_size'          => $settings->{ $prefix . 'button_size' },
 			'button_label'         => $settings->{ $prefix . 'button_label' },
 			'hide_funding_methods' => $settings->{ $prefix . 'hide_funding_methods' },
 			'credit_enabled'       => $settings->{ $prefix . 'credit_enabled' },
 		);
 
+		if ( ! is_array( $data['hide_funding_methods'] ) ) {
+			$data['hide_funding_methods'] = array();
+		}
+
 		$button_layout        = $data['button_layout'];
 		$data['button_size']  = 'vertical' === $button_layout && 'small' === $data['button_size']
 			? 'medium'
 			: $data['button_size'];
 
 		if ( ! wc_gateway_ppec_is_credit_supported() ) {
 			$data['credit_enabled'] = 'no';
-			if ( ! is_array( $data['hide_funding_methods'] ) ) {
-				$data['hide_funding_methods'] = array( 'credit' );
-			} elseif ( ! in_array( 'credit', $data['hide_funding_methods'] ) ) {
+			if ( ! in_array( 'credit', $data['hide_funding_methods'] ) ) {
 				$data['hide_funding_methods'][] = 'credit';
 			}
 		}
 
+		// If we need to create a billing agreement or this is a subscription's single product page, hide credit/debit card options.
+		if (
+			! in_array( 'card', $data['hide_funding_methods'] )
```

Comment:
While not critical, the condition here is missing some parenthesis: it should be `A && (B || C)` but it currently is `A && B || C` which PHP translates to `(A && B) || C` due to operator precedence.
What this means is that "card" gets added twice to the array if it was already in the array *and* you're on a subscription product page (that is, A = false, C = true).